crates/kitsune_p2p/kitsune_p2p/src/test.rs

```diff
@@ -229,62 +116,31 @@ mod tests {
             .await
             .unwrap();
 
+        harness.ghost_actor_shutdown().await?;
+
         assert_eq!(1, res.len());
         for r in res {
             let data = String::from_utf8_lossy(&r.response);
             assert_eq!("echo: test-multi-request", &data);
             assert!(r.agent == a2 || r.agent == a3);
         }
 
-        p2p.ghost_actor_shutdown().await.unwrap();
-        r_task.await.unwrap();
+        Ok(())
     }
 
     #[tokio::test(threaded_scheduler)]
-    async fn test_single_agent_multi_request_workflow() {
-        let space1: Arc<KitsuneSpace> =
-            Arc::new(b"ssssssssssssssssssssssssssssssssssss".to_vec().into());
-        let a1: Arc<KitsuneAgent> =
-            Arc::new(b"111111111111111111111111111111111111".to_vec().into());
+    async fn test_single_agent_multi_request_workflow() -> Result<(), KitsuneP2pError> {
```

Comment:
So if there are other agents online we skip ourself but if not then we will get a response from ourself?
If this is the case we get into an awkward situation when getting a hash that we are the authority for but others are online. It means when we could have done a local call we are guaranteed to hit a network call.
I wonder if we always want to prefer ourself if we are in the dht arc?

Reply:
Personally, I trust us the least - we may have just come online and not had time to sync data yet... hence using us as the fallback. This may be something we want to tweak once we start getting real usage data.